src/backend/dwarf.c

```diff
@@ -100,7 +100,7 @@ void dwarf_addrel(int seg, targ_size_t offset, int targseg, targ_size_t val = 0)
 void dwarf_addrel64(int seg, targ_size_t offset, int targseg, targ_size_t val)
 {
 #if ELFOBJ
-    ElfObj::addrel(seg, offset, R_X86_64_64, MAP_SEG2SYMIDX(targseg), val);
+    ElfObj::addrel(seg, offset, I64 ? R_X86_64_64 : R_386_32, MAP_SEG2SYMIDX(targseg), val);
```

Comment:
Oh, that's a bug I think, dwarf_appreladdr should add an address sized value, so 32-bit on X86.